### DISCUSSION TITLE
OpenJDK vs AdoptOpenJDK

### DISCUSSION THREAD

**Discussion by sigzero** [Homebrew/discussions#1533] [Everyday usage]
If there is a package that lists `OpenJDK` as a requirement, does installing the same version of `AdoptOpenJDK` satisfy that requirement? Or does it have to be `OpenJDK`? Trying to decide which route to use.

**Comment by gromgit** [comment]
`depends_on "XYZ"` means the `XYZ` formula is required, period.

`depends_on :ABC` means `ABC` points to a _Requirement_ class that runs some code to determine if your existing setup meets the necessary prerequisites, and tells you what needs to be done if the requirements aren't fulfilled. For instance, several core formulae specify `depends_on :linux`, which causes `LinuxRequirement` to prevent installs on macOS.
